sopel/plugin.py

```diff
@@ -132,6 +134,45 @@ def add_attribute(function):
     return add_attribute
 
 
+def rules_match(match: MatchText = MatchText.RAW) -> typing.Callable:
```

Comment:
I've a tradeoff then: in a future version, I can craft a system that would replace the flimsy data structure currently used (storing a bunch of string on a callable) that is hard to manipulate whenever we want to attach anything to a pattern itself, by a more mature, more complex, and backward compatible data structure (probably with an object).

That would allow me to add as many properties (i.e. attributes) to a rule pattern without too much trouble. But not for now!

Reply:
@dgw any thoughts about my answer? I still think it's a real PITA to add any property to a rule definition as long as we store them in a list of string/regex object, and I'm very reluctant to change that for Sopel 8. Maybe Sopel 8.1 at best?

Reply:
You know, I think this has languished because it feels like an incomplete solution. You listed some caveats in the description, plus there's this subthread about improving the data structure used for storing patterns to allow setting a match-mode for each instead of the whole function.

How upset would you be if this got tabled until 8.1? I don't see it as critical for the next release, and it would allow time for actually reworking the data structures to make the originally envisioned solution (using keyword-only args) happen.

Reply:
No problem for me! I think it's fair, and a good tradeoff.

Here is what I'll do: I'll change the decorator name and the internal attribute in this PR, as it will cost only ~5min, I'll rebase the branch to keep it up to date, and then I'll close the PR for now.